### PR TITLE
trim fields to match mailer

### PR DESCRIPTION
**Pull request by darinpope**
Trim fields to match similar changes that were made to mailer.

Also matched descriptions to match mailer.

Finally, I added the same form validator to defaultSuffix that exists in mailer.

Signed-off-by: Darin Pope <darin@planetpope.com>

### CODE REVIEW

File: src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java [hunk 1]
```diff
@@ -453,6 +454,13 @@ public void setDefaultContentType(String contentType) {
         }
     }
 
+    public FormValidation doCheckDefaultSuffix(@QueryParameter String value) {
+        if (value.matches("@[A-Za-z0-9.\\-]+") || Util.fixEmptyAndTrim(value)==null)
```
Comment (proxxz) [inline src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:458]: Are underscores not allowed in the suffix?

Comment (darinpope) [reply src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:458]: according to RFC 1035 (...have as interior characters only letters, digits, and hyphen), underscores are not allowed.

This method was copied from mailer-plugin, so if it should be allowed, it probably should be done in both here and in mailer.

**Review by proxxz — APPROVED** [review]

File: src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java (hunk 1 above)
Comment (proxxz) [reply src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:458]: Thank you for the clarification.  The PR looks great to me!

File: src/main/java/hudson/plugins/emailext/MailAccount.java [hunk 2]
```diff
@@ -23,18 +22,18 @@
 
     @Deprecated
     public MailAccount(JSONObject jo){
-        address = nullify(jo.optString("address", null));
-        smtpHost = nullify(jo.optString("smtpHost", null));
-        smtpPort = nullify(jo.optString("smtpPort", null));
+        address = Util.nullify(jo.optString("address", null));
```
Comment (slide) [inline src/main/java/hudson/plugins/emailext/MailAccount.java:25]: This constructor is no longer used, it is there only for binary compatibility.

**Pull request merged**